pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package logger
+
+import (
+	"fmt"
+	"go.uber.org/zap/zapcore"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+type ECSErrorLogger struct {
+	ErrorFilePath    string
+	ErrorLogMaxAge   int
+	ErrorFileMaxSize int
+
+	errorChannel chan zapcore.Entry
+	queue        []zapcore.Entry
+	ticker       time.Ticker
+	errorFile    *os.File
+	size         int
+}
+
+func NewECSErrorLogger() *ECSErrorLogger {
+	ecsErrorFilePath := os.Getenv("STATUS_MESSAGE_FILE_PATH")
```

Comment:
Can you make those env var key constant, also do we need a prefix like `ECS_ERROR_LOG`, the `STAUS_MESSAGE_` seems too general.

Reply:
sure